src/ui/src/js/configs/compiler_config.js

```diff
@@ -0,0 +1,8 @@
+compilerConfig.$inject = ['$compileProvider']
+/**
+ * compilerConfig - Angular configuration object for the Angular compiler.
+ * @param {$compileProvider} $compileProvider Angular's $compileProvider object.
+ */
+export function compilerConfig($compileProvider) {
+    $compileProvider.aHrefSanitizationWhitelist(/^\s*(https?|ftp|mailto|tel|file|blob):/);
```

Comment:
This (or something equivalent) is absolutely necessary, as far as I can tell.

The thing that's being done now that the old code didn't do is exporting a file for download that's created directly from the frontend code (versus downloading a file from a server, for example).

This line essentially adds the "blob" to the whitelist, which is needed because the solution uses JS Blob objects to create the link and the file. There may be other ways that the Angular masters know about to do the same thing that wouldn't require this, but I couldn't find any reference to them (for instance using JS File objects instead).

Evidently creating a new file in the frontend is something that is not a common use case for Angular. At any rate, chrome will refuse to download the file (because of security) if this line is not included.

Reply:
Their deprecation was just a rename, so I followed suit.